src/inventorymanager.cpp

```diff
@@ -272,14 +280,26 @@ void IMoveAction::apply(InventoryManager *mgr, ServerActiveObject *player, IGame
 	} else {
 		// Destination is detached
 		if (to_inv.type == InventoryLocation::DETACHED) {
-			ItemStack src_item = list_from->getItem(from_i);
+			if (fix_replacements) {
+				int count = PLAYER_TO_SA(player)->detached_inventory_AllowTake(
+						to_inv.name, to_list, to_i, replaced, player);
+				if (replaced.count > count)
+					return;
```

Comment:
what's this for?

Reply:
This piece checks whether it is possible to take items out of Detached Inventory (e.g, let Detached inventory be put-only, and have item A. Then putting item B on top of A would return A to player without this)